flink-runtime/src/test/java/org/apache/flink/runtime/io/network/netty/CreditBasedPartitionRequestClientHandlerTest.java

```diff
@@ -419,6 +456,68 @@ public void testNotifyCreditAvailableAfterReleased() throws Exception {
 		}
 	}
 
+	@Test
+	public void testReceivedBufferForRemovedChannel() throws Exception {
+		final int bufferSize = 1024;
+
+		NetworkBufferPool networkBufferPool = new NetworkBufferPool(10, bufferSize, 2);
+		SingleInputGate inputGate = createSingleInputGate(1);
+		RemoteInputChannel inputChannel = createRemoteInputChannel(inputGate, null, networkBufferPool);
```

Comment:
Use this way instead? Then we do not rely on `null` `PartitionRequestClient`, and I guess `createRemoteInputChannel` is mainly for indicating the required `PartitionRequestClient`.
```
InputChannelBuilder.newBuilder()
.setMemorySegmentProvider(networkBufferPool)
.buildRemoteAndSetToGate(inputGate);
```

Reply:
Changed to using the builder.